### ISSUE TITLE
Firefox 46 on Linux brings GTK3 compatibility and shows discrepencies in MODX manager theme.

### ISSUE THREAD

**Issue by muzzwood** [modxcms/revolution#12991]
### Summary

MODX Manager theme relies on browser defaults for scroll bars and form input text. With new Firefox, those defaults are not what they used to be. In the interest of uniformity across platforms, default colours for form text and scroll bars should be specified in CSS.
### Step to reproduce

Open the MODX manager on any Linux desktop environment with a dark theme.
### Observed behavior

See attached screenshots
### Expected behavior

Form text should be readable. Scroll bars should not be dark.
### Environment

MODX 2.5.0. Ubuntu Gnome 15.10. Firefox 46.0

If you look closely you can see there is white text in the username field
[http://imgur.com/ZATRkNn](http://imgur.com/ZATRkNn)

Note the dark scroll bar
[http://imgur.com/1XMhjnL](http://imgur.com/1XMhjnL)

**Comment by Jako** [comment]
In my eyes it is more a Firefox than a MODX issue. So please tell us, why those default colors should not be specified in the internal browser css.

**Comment by digitalpenguin** [comment]
You're right, it's a Firefox issue. But it means that the MODX manager is now unusable on this platform. With a couple of lines of CSS, it could be made usable.

**Comment by Mark-H** [comment]
Yeah that should get fixed, especially the input text. Is it like that on every form across the manager?

**Comment by digitalpenguin** [comment]
Yes, every input/textarea field as far as I can see

**Comment by Jako** [comment]
Hmm, there are no reset css rules in https://github.com/modxcms/revolution/tree/2.x/_build/templates/default/sass
where this would belong to.

**Comment by digitalpenguin** [comment]
I'm not sure of the order the files are compiled but what about here: 
https://github.com/modxcms/revolution/blob/2.x/_build/templates/default/sass/_colors-and-vars.scss
?

**Comment by jaygilmore** [comment]
I wonder what @jpdevries has to say on this matter? While a seamless app experience is desirable caution may need to be taken to alter a user's browser settings.

**Comment by Mark-H** [comment]
Is there really a question wether white-on-white text is bad? It's unusable, that's a bug.

**Comment by Jako** [comment]
It is a bug of Firefox (or Gnome) that switches the text color to white when some theme is changed to dark.

**Comment by digitalpenguin** [comment]
I've also put in a bug report for Firefox. If people are uncomfortable with setting a default input text colour (I don't see why...), perhaps Mozilla will fix the issue quickly.

**Comment by jaygilmore** [comment]
Is the `background-color` set in the inputs and textareas in MODX? If not, then it's a bug in MODX. If they are set and Firefox's settings are overriding the background but not inversing the text, that's a Firefox bug. Background color should be set via CSS for all input fields because you cannot ever predict what the user will do with it.

**Comment by jaygilmore** [comment]
Looking at the Manager CSS, it's obvious why this is happening. The CSS is being applied to the ExtJS classes and not to the input element itself. My guess is that Firefox is overriding the input background-color because there is no color set for the element. 

The way it's set is as below: 
```css
.x-form-text, textarea.x-form-field, .x-form-textarea {
    background-color: #FBFBFB;
    background-image: none;
    border-radius: 3px;
    border: 1px solid #E4E4E4;
    position: relative;
    transition: border-color 0.25s;
}```

**Comment by sottwell** [comment]
A classname should override an element name.

**Comment by sottwell** [comment]
https://developer.mozilla.org/en/docs/Web/CSS/Specificity

**Comment by jaygilmore** [comment]
By spec, yes. But It depends on how this theme applied in the browser and what specificity the browser maker gave it. They may have felt that it was important to the user to apply the style heavily.

**Comment by sottwell** [comment]
I don't see how any web developer, MODX or otherwise, is supposed to be able to guess what OS or browser will override their page CSS.

**Comment by jaygilmore** [comment]
Whoops, it seems that I made a mistake. This has nothing to do with the background but to do with fact there is no `color` setting on the inputs by class nor on the element itself. All the browser can do is assume the browser is in charge. This is a MODX bug. Sorry for the red herring.

**Comment by rthrash** [comment]
red herring from smashingred … go figure :P

**Comment by sottwell** [comment]
Yeah, looking again at that image, the background color is what we would expect. It's the text that is inheriting the color set for the body.

**Comment by Jako** [comment]
```
.x-form-element input[type=text], .x-form-element input[type=password], .x-form-element textarea { color: #000 }
```

**Comment by Jako** [comment]
@sottwell In Firefox the user agent stylesheet is changed from black to white (by color: -moz-FieldText; somewhere in resource://gre-resources/forms.css). 

The color of the body is #081922. So only an explizit styling of text inputs and textareas would help. But a webpage is a container that should not be changed from external without thinking about the consequences.

**Comment by jaygilmore** [comment]
The way I've always approached it and seen it approached is to explicitly declare color and background for each element  if you want to have a clear foundation that won't vary from browser to browser. Elements are fair game. Browser chrome objects such as scrollbars etc, should be handled with extreme care.

**Comment by Jako** [comment]
Then we have to add some normalize css in the manager css.

**Comment by Ibochkarev** [comment]
@digitalpenguin is this request relevant?

**Comment by muzzwood** [comment]
Not so much anymore. 
It might be relevant in the future again if a browser changes the way fields are rendered since we haven't added normalizing css yet. But I'll close it for now, perhaps we'll never need to worry about it again. Cheers! ;)

**Comment by Ibochkarev** [comment]
I'll try adding normalize.css. Thanks

**Comment by muzzwood** [comment]
Dark scrollbars in Firefox on Linux are still an issue, so I'm re-opening this.

The fix is to add `scrollbar-color: #e1e3e6 rgba(0, 0, 0, 0);` to the body selector in index.scss
https://github.com/modxcms/revolution/blob/3.x/_build/templates/default/sass/index.scss#L56-L59

Note that `scrollbar-color` is only supported by Firefox, so it won't affect other browsers.
The color chosen above is just what Chromium uses in the same environment.

**Comment by Ibochkarev** [comment]
Fixed #15962
